gradunwarp/core/unwarp_resample.py

```diff
@@ -199,167 +202,174 @@ def non_linear_unwarp_siemens(self, volshape, dv, dxyz, m_rcs2lai, m_rcs2lai_noh
         vc, vr = utils.meshgrid(np.arange(nc), np.arange(nr))
 
         # Compute transform to map the internal voxel coordinates to FSL scaled mm coordinates
-        try:
-            pixdim1=float((subprocess.Popen(['fslval', self.name,'pixdim1'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        except ValueError:
-            log.error('Failure during fslval call. Make sure fslval, fslhd, fslorient are in your PATH, and that FSLOUTPUTTYPE is set.')
-            sys.exit(1)
-        
-        pixdim2=float((subprocess.Popen(['fslval', self.name,'pixdim2'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        pixdim3=float((subprocess.Popen(['fslval', self.name,'pixdim3'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        dim1=float((subprocess.Popen(['fslval', self.name,'dim1'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        outputOrient=subprocess.Popen(['fslorient', self.name], stdout=subprocess.PIPE).communicate()[0].strip()
-        if outputOrient == b'NEUROLOGICAL':
-            log.info('Input volume is NEUROLOGICAL orientation. Flipping x-axis in output fullWarp_abs.nii.gz')
+        pixdim1, pixdim2, pixdim3 = self.input_nii.header.get('pixdim')[1:4]
+
+        dim1 = self.input_nii.header.get('dim')[1]
+
+        # TODO: figure out if that logic works
+        outputOrient = get_fslorient(self.input_nii)
+        if outputOrient == b"NEUROLOGICAL":
```

Comment:
Simpler:

```suggestion
        if np.linalg.det(self.input_nii.affine) > 0:
```

You don't really care about all the logic of consistent or inconsistent or unknown. You just need the affine according to the NIfTI 3 methods, which both FSL and nibabel implement. And this also doesn't tie you to NIfTI. It will work just fine with an MGH file.

Reply:
It might be cleaner to restrict it to the 3x3 determinant - the 4th row being 0 0 0 1 (as it always should be, but we don't check that here) should cause it to ignore the translation values, but I prefer not to rely on assumptions when I don't have to.